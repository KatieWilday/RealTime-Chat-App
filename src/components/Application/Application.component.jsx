import React from "react";
import { auth } from "../../firebase/firebase.utils";

import Messages from "../Messages/Messages.component";
import Auth from "../Auth/Auth.component";

import "./Application.styles.scss";

class Application extends React.Component {
  state = {
    user: null
  };

  unsubscribeFromAuth = null;

  componentDidMount = async () => {
    document.body.style.background = "#333";
    this.unsubscribeFromAuth = auth.onAuthStateChanged(user => {
      this.setState({ user });
    });
  };
  componentWillUnmount = () => {
    document.body.style.background = "white";
  };
  render() {
    return (
      <div>
        <Auth user={this.state.user} />
        <Messages />
      </div>
    );
  }
}
export default Application;
