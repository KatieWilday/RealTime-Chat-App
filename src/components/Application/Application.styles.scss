.main-bg {
  background: -webkit-radial-gradient(
    top,
    circle cover,
    #00d2ff 0%,
    #3a7bd5 80%
  );
  height: 100vh;
}
