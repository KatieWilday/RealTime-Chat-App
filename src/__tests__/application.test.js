import React from "react";
import { create } from "react-test-renderer";
import Application from "../components/Application/Application.component";

it("snapshot", () => {
  const tree = create(<Application />).toJSON();
  expect(tree).toMatchSnapshot();
});
