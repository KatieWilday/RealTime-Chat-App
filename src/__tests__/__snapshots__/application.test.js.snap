// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <div
    className="sign-in-and-sign-up"
  >
    <div
      className="sign-in"
    >
      <h2
        className="title"
      >
        I already have an account
      </h2>
      <span>
        Sign in with your email and password
      </span>
      <form
        onSubmit={[Function]}
      >
        <div
          className="group"
        >
          <input
            className="form-input"
            name="email"
            onChange={[Function]}
            required={true}
            type="email"
            value=""
          />
          <label
            className=" form-input-label"
          >
            Email
          </label>
        </div>
        <div
          className="group"
        >
          <input
            className="form-input"
            name="password"
            onChange={[Function]}
            required={true}
            type="password"
            value=""
          />
          <label
            className=" form-input-label"
          >
            Password
          </label>
        </div>
      </form>
      <div
        className="login-button"
        onClick={[Function]}
      >
        <div
          className="icons-top envelope"
        >
          <i
            className="fa fa-envelope fa-2x"
          />
        </div>
        <div
          className="login-button-text"
        >
          Sign-in
           With Email
        </div>
      </div>
      <div
        className="custom-button gplus"
        onClick={[Function]}
      >
        <div
          className="icons fa-gplus"
        >
          <i
            className="fa fa-google"
          />
        </div>
        <div
          className="signtext"
        >
          Sign-
          in
           with Gmail
        </div>
      </div>
    </div>
    <div
      className="sign-up"
    >
      <h2
        className="title"
      >
        I do not have an account
      </h2>
      <span>
        Sign up with your email and password
      </span>
      <form
        className="sign-up-form"
        onSubmit={[Function]}
      >
        <div
          className="group"
        >
          <input
            className="form-input"
            name="displayName"
            onChange={[Function]}
            required={true}
            type="text"
            value=""
          />
          <label
            className=" form-input-label"
          >
            Display Name
          </label>
        </div>
        <div
          className="group"
        >
          <input
            className="form-input"
            name="email"
            onChange={[Function]}
            required={true}
            type="text"
            value=""
          />
          <label
            className=" form-input-label"
          >
            Email
          </label>
        </div>
        <div
          className="group"
        >
          <input
            className="form-input"
            name="password"
            onChange={[Function]}
            required={true}
            type="password"
            value=""
          />
          <label
            className=" form-input-label"
          >
            Password
          </label>
        </div>
      </form>
      <div
        className="login-button"
        onClick={[Function]}
      >
        <div
          className="icons-top envelope"
        >
          <i
            className="fa fa-envelope fa-2x"
          />
        </div>
        <div
          className="login-button-text"
        >
          Sign-up
           With Email
        </div>
      </div>
      <div
        className="custom-button gplus"
        onClick={[Function]}
      >
        <div
          className="icons fa-gplus"
        >
          <i
            className="fa fa-google"
          />
        </div>
        <div
          className="signtext"
        >
          Sign-
          up
           with Gmail
        </div>
      </div>
    </div>
  </div>
</div>
`;
